tls/message.py

```diff
@@ -49,16 +49,8 @@ def as_bytes(self):
                 certificate_types=[cert_type.value
                                    for cert_type in self.certificate_types]
             ),
-            supported_signature_algorithms=Container(
-                supported_signature_algorithms_length=2 * len(
-                    self.supported_signature_algorithms
-                ),
-                algorithms=[Container(
-                    hash=algorithm.hash.value,
-                    signature=algorithm.signature.value,
-                )
-                    for algorithm in self.supported_signature_algorithms
-                ]
+            supported_signature_algorithms=ListContainer(
```

Comment:
[The Travis builds fail with a TypeError](https://travis-ci.org/pyca/tls/jobs/176333212#L268) because our `SignatureAndHashAlgorithm` class doesn't implement `Container`'s attribute-or-getitem interface.  I believe the solution is to wrap each `SignatureAndHashAlgorithm` in a `Container`, like so:

```python
            supported_signature_algorithms=[
                Container(hash=sa.hash, signature=sa.signature)
                for sa in self.supported_signature_algorithms
            ],
```

(You can wrap this in a `ListContainer`, which seems to be a `list` with a pretty-printing `__str__` implementation)